### DISCUSSION TITLE
Broken Python Examples

### DISCUSSION THREAD

**Discussion by Cambyst** [perspective-dev/perspective#2269] [Q&A] [answered]
## Bug Report
Hi, new to Perspective and it looks awesome. However it seems that a change in Perspective python import causing all examples and docs to be outdated.

### Steps to Reproduce:
 1. Run any of the server.py under the python-* examples

### Expected Result:
Expected the server.py to run

### Actual Result:
ImportError: cannot import name 'Table' from 'perspective'

### Environment:
Python 3.10
perspective-python 2.0.1 (installed via pip)

### Additional Context:

Could it be caused by the following ?

https://github.com/finos/perspective/commit/2cf241f1398791705ccb6f57838bed60163e7e11

**Comment by timkpaine** [comment]
Can you provide a stack trace of the error

**Comment by Cambyst** [comment]
Hi @timkpaine , thanks for checking! Here is an example for python-tornado, but same apply to all examples (including the one in python docs):

![image](https://github.com/finos/perspective/assets/84931463/70045910-c3f1-4bc7-ad34-4988ac6c3886)

**Comment by texodus** [comment]
@Cambyst It looks like you're on a mac, what is your OSX version and CPU (x85 or arm64)?

@timkpaine wants the errors from your installation log, not the import statement.

**Comment by Cambyst** [comment]
Hey @texodus , I am on macOS 13.4 with an Apple M1 (arm64).

Attached is install log and also first import log on a clean python env. It seems to confirm your intuition that it is due to the CPU architecture.
[first_import.txt](https://github.com/finos/perspective/files/11843393/first_import.txt)
[install_log.txt](https://github.com/finos/perspective/files/11843394/install_log.txt)

**Comment by texodus** [comment]
You're not installing `perspective-python` 2.0.1 as you say in your report - you're installing 2.3.0 as printed in your `install_log.txt`.  This version has a regression in arm64 build, you can downgrade to 2.2.1 (or 2.0.1 incidentally), or wait until 2.3.1.

**Comment by Cambyst** [comment]
Thank you @texodus , you are right that on the clean env, it installed 2.3 and that downgrading to 2.2.1 helped a little (not sure why pip-show print 2.0.1 originally but nevermind). However on launch it is still throwing an exception (I have run "yarn install", although this was not explicitely documented):

![image](https://github.com/finos/perspective/assets/84931463/e513f615-8add-41ca-8680-41c99d899cb9)

**Reply by timkpaine** [reply]
looks like you’re trying to run the examples from master, but you installed perspective 2.2.1. You’ll need to run the examples as of perspective 2.2.1

**Comment by Cambyst** [comment]
Apologies for my lack of knowledge around Yarn/Javascript, but it seems that none of the "/dist/cdn" script referenced in the html files are also not installed by Yarn. Example if python-tornado/index.html:

https://github.com/finos/perspective/blob/46809f35c43a39d3e3c4838a28ce159715271623/examples/python-tornado/index.html#L15

**Reply by timkpaine** [reply, marked as answer]
these are served locally from the build JS assets. `yarn build_js` to build them, the examples are designed to be run from a locally built developer installation (or you can also easily adapt them to a public CDN like jsdelivr)

**Comment by Cambyst** [comment]
Works like a charm now, thank you very much for your time @timkpaine @texodus , much appreciated !
